src/main.rs

```diff
@@ -63,140 +64,146 @@ fn response_function(req: Request<Body>) -> BoxFut {
     // Process input api path
     let parameters = common::get_restful_parameters(req.uri().path());
 
-    match req.method() {
-        &Method::GET => {
-            info!("GET invoded");
-
-            // invalid requestd handling
-            if parameters.is_empty() {
-                if let Err(e) = common::set_response_content(
-                    200,
-                    "Not Implemented: Use /v2/keys/ or /v2/quotes/ interfaces",
-                    Map::new(),
-		    &mut my_response,
-                ) {
-		    error!("Failed with error code {}.", e);
-		}
-            } else if parameters.contains_key("keys") {
-                let mut response_map = Map::new();
-
-                match parameters.get(&"keys") {
-                    // check Kb value is available to perform the do_hmac
-                    // crypto request verify will do hmac for the challenge
-                    // orignal function: crypto.do_hmac(self.server.K,
-                    // challenge)
-                    Some(&"verify") => {
-                        /* /keys/verify/challenge/blablabla */
-                        let challenge = parameters.get(&"challenge");
-                        let result = crypto::do_hmac(
-                            common::KEY.to_string(),
-                            challenge.unwrap().to_string(),
-                        );
-                        match result {
-                            Ok(hmac) => {
-                                info!(
-                                    "
+    // Loop scope wrap around the request handling
+    // Exit: Encounter error early exit or exit at the end to the scope
+    loop {
+        match req.method() {
```

Comment:
I think it might be more clear to have a function for actual request handling that returns a result, then loop-and-log around that.

Reply:
Create two new functions
`get_request_handler()` and `post_request_handler()'
which return Result for error handling.
In such kind of implementation, no loop scope is needed anymore.